test-dts/mount.d-test.ts

```diff
@@ -13,12 +13,23 @@ const AppWithDefine = defineComponent({
   template: ''
 })
 
-// accept props
-let wrapper = mount(AppWithDefine, {
-  props: { a: 'Hello', b: 2 }
-})
-// vm is properly typed
-expectType<string>(wrapper.vm.a)
+// accept props- vm is properly typed
+expectType<string>(
+  mount(AppWithDefine, {
+    props: { a: 'Hello', b: 2 }
+  }).vm.a
+)
+
+// no data provided
+expectError(
+  mount(AppWithDefine, {
+    data() {
+      return {
+        myVal: 1
+      }
+    }
+  })
+)
```

Comment:
👍 I think more tests with data would be awesome. Can we pass `data` to all type of components? Can we pass data that are not declared? etc. That will greatly help us maintining the lib if we have extensive coverage of the different cases.